controller/api/public/grpc_server_test.go

```diff
@@ -174,3 +187,84 @@ spec:
 		}
 	})
 }
+
+func listServiceResponsesEqual(a pb.ListServicesResponse, b pb.ListServicesResponse) bool {
+	if len(a.Services) != len(b.Services) {
+		return false
+	}
+
+	sort.Sort(ByService(a.Services))
+	sort.Sort(ByService(b.Services))
+
+	for i := 0; i < len(a.Services); i++ {
+		aSvc := a.Services[i]
+		bSvc := b.Services[i]
+
+		if aSvc.Name != bSvc.Name || aSvc.Namespace != bSvc.Namespace {
+			return false
+		}
+	}
+
+	return true
+}
+func TestListServices(t *testing.T) {
+	t.Run("Successfully performs a query based on resource type", func(t *testing.T) {
```

Comment:
Super tiny nit: I think this string was copied from `TestListPods`, but it's not quite right in this context. Maybe it could be "Successfully queries for services" or something similar?

Reply:
ah yeah totally!